WordPress/gutenberg
# Remove `role` attribute when set to `null` in `data-wp-bind`

## What?
In this pull request I'm adding a workaround to handle the `role` attribute properly until it is solved in Preact. Basically, when we pass a `null` value to `role`, it is using and empty string instead of removing the HTML element. I submitted [this issue](https://github.com/preactjs/preact/issues/4136) to Preact repo.

As part of this, I also removed the Server Side Rendering attributes to fix the issue reported [here](https://github.com/WordPress/gutenberg/issues/54560). After this fix it seems that the `aria-modal` and `role` attributes are properly removed in the Server Side Rendering and after opening and closing the menu.

I didn't add logic to remove the `aria-label` because it was there before adding the Interactivity API, and some users may be using that. Additionally, as reported in the issue, it shouldn't affect the accessibility.

## Why?

Right now, there is no way to remove the `role` attribute dynamically, which ends up in an empty value, which is not valid HTML.

## How?
I added a workaround to handle the `role` attribute until the issue in Preact is solved. Additionally, I removed the attributes for the Server Side Rendering.

## Testing Instructions

1. Include a navigation block in your theme.
2. Go the the frontend, inspect the navigation menu, and check that the `div` element with class `wp-block-navigation__responsive-dialog` doesn't have `aria-modal` or `role` attributes.
3. Open the menu and check that the same `div` has `aria-modal="true"` and `role="dialog"`.
4. Close the menu and check that both attributes are removed.
